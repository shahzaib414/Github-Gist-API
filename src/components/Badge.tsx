import React from "react";
import styled from "styled-components";

interface Props {
    text: string;
    color?: string;
}

const StyledDiv = styled.div<{ color: string }>`
    padding: 10px;
    margin-right: 5px;
    display: flex;
    border-radius: 50px;
    background-color: ${(props) => props.color};
    color: white;
    font-size: 14px;
`;

const Badge = ({ color, text }: Props) => {
    return <StyledDiv color={color || "#4118C9"}>
        {text}
    </StyledDiv>
}

export default Badge;