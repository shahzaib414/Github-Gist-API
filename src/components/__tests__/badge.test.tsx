import React from 'react';
import renderer from 'react-test-renderer';
import Badge from "../badge";

test("Badge Component renders correctly", () => {
    const component = renderer.create(<Badge text="Typescript"/>)
    expect(component).toMatchSnapshot();
})
