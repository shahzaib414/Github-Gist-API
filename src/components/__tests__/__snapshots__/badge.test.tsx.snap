// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders correctly 1`] = `
<div
  className="sc-bdvvaa fGlDwO"
  color="#4118C9"
>
  Typescript
</div>
`;
